src/main/java/org/opentripplanner/common/model/T3.java

```diff
@@ -0,0 +1,59 @@
+package org.opentripplanner.common.model;
+
+import java.io.Serializable;
+
+/**
+ * An ordered object of three (potentially different types)
+ */
+public class T3<E1, E2, E3> implements Serializable {
+  private static final long serialVersionUID = 1L;
+
+  public final E1 first;
+
+  public final E2 second;
+
+  public final E3 third;
+
+  public T3(E1 first, E2 second, E3 third) {
+    this.first = first;
+    this.second = second;
+    this.third = third;
+  }
+
+  @Override
```

Comment:
If you convert this to a `record` then hashCode, equals and toString are generated for you.

Reply:
Sounds like a good idea - will do that